src/components/Collections/CollectionShare/publicShare/manage/linkExpiry.tsx

```diff
@@ -23,30 +24,71 @@ export function ManageLinkExpiry({
     const updateDeviceExpiry = async (optionFn) => {
         return updatePublicShareURLHelper({
             collectionID: collection.id,
-            validTill: optionFn(),
+            validTill: optionFn,
         });
     };
+    const [shareExpiryOptionsModalView, setShareExpiryOptionsModalView] =
+        useState(false);
+    const [shareExpiryValue, setShareExpiryValue] = useState(0);
+    const [labelText, setLabelText] = useState(
+        publicShareProp?.validTill
+            ? formatDateTime(publicShareProp?.validTill / 1000)
+            : 'never'
+    );
+    useEffect(() => {
+        if (shareExpiryOptionsModalView) {
+            setShareExpiryOptionsModalView(true);
+        } else setShareExpiryOptionsModalView(false);
+    }, [shareExpiryOptionsModalView]);
+    const closeShareExpiryOptionsModalView = () =>
+        setShareExpiryOptionsModalView(false);
+    const openShareExpiryOptionsModalView = () =>
+        setShareExpiryOptionsModalView(true);
+    const changeshareExpiryValue = (value: number) => () => {
+        updateDeviceExpiry(value);
+        setLabelText(
+            publicShareProp?.validTill
+                ? formatDateTime(publicShareProp?.validTill / 1000)
+                : 'never'
+        );
+        setShareExpiryValue(value);
+        shareExpiryValue;
+        setShareExpiryOptionsModalView(false);
+    };
```

Comment:
The code here (line 30 to 57) is looking super cluttered, Normally you should separate different types of stuff with empty line gaps

Like all declarations, can be together as one group, then each useEffect (any types functions in general)  should have an empty line before and after it. 

Improves code readability a lot